debugging.md

```diff
@@ -0,0 +1,52 @@
+# Debugging Go based operator
+
+## Debugging operator in vscode
+
+### Prerequisite
+
+* install [current used go version](https://github.com/openstack-k8s-operators/install_yamls/blob/master/devsetup/vars/default.yaml)
+* install [current used operator-sdk version](https://github.com/openstack-k8s-operators/install_yamls/blob/master/devsetup/vars/default.yaml)
+* install vscode
+* make sure [go extension for vscode](https://marketplace.visualstudio.com/items?itemName=golang.Go)
+is installed
+* install [delve debugger for go](https://github.com/go-delve/delve)
+```bash
+go install github.com/go-delve/delve/cmd/dlv@latest
+```
+* Deploy the needed operator CRDs/manifests either manually or install and uninstall via olm
+```bash
+oc login -u kubeadmin -p '12345678' https://api.ostest.test.metalkube.org:6443
+make install
+```
+
+### Run debugger via vscode
+A `.vscode/launch.json` needs to be created which instructs vscode how to start the debugger. Create the `.vscode/launch.json` with `Run` -> `Add configurarion` -> `Go: Launch Package`
+```json
+{
+        "version": "0.2.0",
+        "configurations": [
+        
+                {
+                        "name": "Launch Package",
+                        "type": "go",
+                        "request": "launch",
+                        "mode": "auto",
+                        "program": "<path to operator base dir>",
```

Comment:
Shouldn't that be:
`program": "<path to operator base dir>/main.go",`?

Also we should probably add the line
`            "cwd": "<path to operator base dir>",
`
or, what I personally use:
`            "cwd": "${fileDirname}",`

I am not sure if the relative path for OPERATOR_TEMPLATES below would work otherwise, if the .vscode project is not in the operator's root directory itself.

Reply:
> Shouldn't that be: `program": "<path to operator base dir>/main.go",`?
> 
referencing the main package folder works for me as mentioned in https://vscode-debug-specs.github.io/go/ , `program must be main package folder or *.go file`
But seems the information is different on different sites. https://code.visualstudio.com/docs/editor/debugging has `executable or file to run when launching the debugger`

> Also we should probably add the line ` "cwd": "<path to operator base dir>",` or, what I personally use: ` "cwd": "${fileDirname}",`
> 
> I am not sure if the relative path for OPERATOR_TEMPLATES below would work otherwise, if the .vscode project is not in the operator's root directory itself.

good point, will add it